server/tests/Api.IntegrationTests/controllers/CourseControllerTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Net;
+using System.Net.Http.Json;
+using Microsoft.AspNetCore.Mvc.Testing;
+using StudyZen.Application.Dtos;
+
+namespace Api.IntegrationTests.controllers;
+
+public class CourseControllerTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly HttpClient _httpClient;
+    private readonly CreateCourseDto _createCourseDto;
+    private readonly UpdateCourseDto _updateCourseDto;
+
+    public CourseControllerTests(WebApplicationFactory<Program> factory)
+    {
+        _httpClient = factory.CreateClient();
+        _createCourseDto = new CreateCourseDto("Test name", "Test desc");
+        _updateCourseDto = new UpdateCourseDto("New name", "New desc");
+    }
+
+    [Fact]
+    public async Task CreateCourse()
```

Comment:
Test naming should be more explicit, for example, CreateCourse_ValidRequest_ReturnSuccess or something that would have a more descriptive name

Reply:
I could add a longer description, however, since I'm not testing a single outcome, but both http status code and the object that the method created, it would become rather long. The most accurate naming I could think of was CreateCourse_ValidInput_CreatesACourse, so I chose to simplify it and leave just the first part. Alternatively, I could mention only the http return in the naming (like CreateCourse_ValidInput_ReturnsCreated), but then all other methods would basically be the same with ValidInput_ReturnsOk ending. Should I still change it or leave it as is?

Reply:
I thought of another naming option: Should _create_course. Would this be more fitting?

Reply:
In my opinion, `CreateCourse_ValidInput_ReturnsCreated` would be a proper and clear name, having more characters doesn't cost us and gives better readability